### PR TITLE
Add DatePickerAndroid API

### PR DESCRIPTION
**Pull request by tonyhb**
Adds the DatePickerAndroid API bindings.

### CODE REVIEW

File: src/datePickerAndroid.re [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+[@bs.scope "DatePickerAndroid"] [@bs.module "react-native"]
+external _open : optsJs => Js.Promise.t(responseJs) =
+  "open";
+
+let open_ = (~date: Js.Date.t, ~minDate=?, ~maxDate=?, ~mode=Default, ()) =>
+  _open({
+    "date": date,
+    "minDate": Js.Undefined.from_opt(minDate),
```
Comment (arnarthor) [inline src/datePickerAndroid.re]: I'd go with `Js.Nullable` here it covers both null and undefined but the from_opt value get's compiled down to `undefined`

File: src/datePickerAndroid.re [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      | Calendar => "calendar"
+      | Spinner => "spinner"
+      }
+  })
+  |> Js.Promise.then_(
+       (resp: responseJs) => {
+         let an = resp |> action;
+         Js.Promise.resolve(an)
+       }
+     );
```
Comment (arnarthor) [inline src/datePickerAndroid.re]: This is real nitpicky, but I like this style of Promise based flow better 🙈 

```ocaml
Js.Promise.(
  _open({
    "date": date,
    "minDate": Js.Undefined.from_opt(minDate),
    "maxDate": Js.Undefined.from_opt(maxDate),
    "mode":
      switch mode {
      | Default => "default"
      | Calendar => "calendar"
      | Spinner => "spinner"
      }
  })
  |> then_((resp: responseJs) => resp |> action |> resolve)
);
```

File: src/datePickerAndroid.re (hunk 1 above)
Comment (arnarthor) [reply src/datePickerAndroid.re]: I'm also pretty sure `Js.Undefined` and `Js.Null` are getting deprecated in favour of `Js.Nullable` soon

File: src/datePickerAndroid.rei [hunk 3]
```diff
@@ -0,0 +1,24 @@
+[@bs.scope "DatePickerAndroid"] [@bs.module "react-native"] external dateSet : string =
```
Comment (wokalski) [inline src/datePickerAndroid.rei]: The convention in this database is to put externals as let bindings in `rei`s. So in this case it'd be `let dateSet : string;` etc.

Comment (wokalski) [inline src/datePickerAndroid.rei]: `externals` are exposed in `rei`s as let bindings. In this case it'd be `let dateSet : string;`

File: src/datePickerAndroid.re (hunk 2 above)
Comment (tonyhb) [reply src/datePickerAndroid.re]: :) thanks!

File: src/datePickerAndroid.rei (hunk 3 above)
Comment (tonyhb) [reply src/datePickerAndroid.rei]: thanks :)

**Comment by tonyhb** [comment]
Thanks for the updates!  Just heading back from Thanksgiving - will amend and push tonight or tomorrow AM

**Comment by arnarthor** [comment]
@tonyhb did you forget about this one 😄

**Comment by tonyhb** [comment]
Haha sorry, got busy at work!  Jumping on it now

**Comment by tonyhb** [comment]
Fixed - didn't rebase so that the diff is visible.  We can rebase when it merges if necessary. :)

Thanks again for the input - good to learn about externals in REI for example.  Sorry it took a while!

**Comment by tonyhb** [comment]
Should I move reason-react from a peer dependency or change the CI script to explicitly install it?

**Comment by arnarthor** [comment]
No we don't want to move it from peer dependencies. It's essential for Reason projects to not have hard depedencies.

Fixing the CI script would be better

**Comment by MoOx** [comment]
See #105 for resolution for the issue you are facing. I think #105 should be merged asap, and then this PR could be rebased on master.

**Comment by MoOx** [comment]
A fix landed in master, @tonyhb you can just rebase and it should be good.

**Comment by wokalski** [comment]
JS sources are still missing. The rest is great!

**Comment by wokalski** [comment]
@arnarthor @MoOx this one is done, too.

**Comment by wokalski** [comment]
@arnarthor please remove the last merge commit

**Comment by wokalski** [comment]
nvm, I'll take care of it

**Comment by wokalski** [comment]
@arnarthor it's ready.

**Comment by wokalski** [comment]
In my usual fashion I took this one a bit too far. I'll revert 0a0f944

**Comment by arnarthor** [comment]
It can't automatically merge 🤔 do you know why?

**Comment by wokalski** [comment]
you have to accept it 👍

**Comment by arnarthor** [comment]
Ahh right, I accepted it before you reverted a commit 👍

**Pull request merged**